app/routes/metadata.rb

```diff
@@ -1,28 +1,30 @@
+require_relative '../../x'
+
 module ExercismWeb
   module Routes
     class Metadata < Core
-      get '/exercises/:track_id/:slug' do |track_id, slug|
-        status, response = Xapi.get("v2", "exercises", track_id, slug)
-        data = JSON.parse(response)
-        if status == 404
-          flash[:notice] = data['error']
+      get '/exercises/:id/:slug/tests' do |track_id, slug|
+        exercise = X::Exercise::TestFiles.find(track_id, slug)
+        if exercise.files.empty?
+          flash[:notice] = error_message(exercise)
           redirect '/'
```

Comment:
Not a partial a new page, I was thinking about.

Ok so we could move all the fetch and links to a partial to avoid duplication ?? In both pages .